lib/ansible/modules/blockinfile.py

```diff
@@ -306,9 +306,7 @@ def main():
     lines[n0:n0] = blocklines
 
     if lines:
-        result = b('\n').join(lines)
```

Comment:
This code ensured that the block text always had a line separator after it. With this removed, a block that has no line separator ends up being concatenated to the end mark.

Reply:
Sorry I missed that in review.